cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -367,9 +372,180 @@ impl From<ProcessComposeLogContents> for LoggedError {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct ProcessComposeLogLine {
+    pub process: String,
+    pub message: String,
+}
+
+impl ProcessComposeLogLine {
+    /// Construct a new log line.
+    fn new(process: impl Into<String>, message: impl Into<String>) -> Self {
+        Self {
+            process: process.into(),
+            message: message.into(),
+        }
+    }
+}
+
+pub struct ProcessComposeLogStream {
+    readers: Vec<ProcessComposeLogReader>,
+    rx: Receiver<ProcessComposeLogLine>,
+}
+
+impl ProcessComposeLogStream {
+    /// Create a new log stream by attaching to the logs of multiple processes.
+    ///
+    /// For each `process` in `processes`, a new [ProcessComposeLogReader] will be started,
+    /// which will read log lines for the process and send them via MPSC to the receiver.
+    /// [ProcessComposeLogStream] implements [Iterator]
+    /// that will wait for log lines from any of the processes.
+    pub fn new(
+        socket: impl AsRef<Path>,
+        processes: impl IntoIterator<Item = impl AsRef<str>>,
+    ) -> Result<ProcessComposeLogStream, ServiceError> {
+        let (tx, rx) = std::sync::mpsc::channel();
```

Comment:
nit: I would prefer longer names than `tx` and `rx`

Reply:
these are the common names used in the [std lib docs](https://doc.rust-lang.org/stable/std/sync/mpsc/index.html#examples) but I can change to receiver sender instead to make it clearer👍🏼

Reply:
When I was reviewing I looked at the docs for `channel()` and they use sender and receiver https://doc.rust-lang.org/stable/std/sync/mpsc/fn.channel.html

Reply:
gotcha, changed them to sender and receiver